front/components/dialogs/DefaultDialog.vue

```diff
@@ -0,0 +1,84 @@
+<!-- eslint-disable vue/no-v-html -->
+<template>
+  <v-dialog v-model="show" content-class="modal" persistent max-width="600">
+    <div class="closeBtn">
+      <v-icon text class="cancelIcon" @click="$emit('hide')">mdi-close</v-icon>
+    </div>
+
+    <slot name="header">
+      <!-- Header -->
+      <section class="modalHeader">
+        <h3>{{ header }}</h3>
+        <v-img
+          v-if="imagePath"
+          max-height="180"
+          max-width="180"
+          class="mx-auto modalTopImage"
+          contain
+          :src="imagePath"
+        ></v-img>
+      </section>
+    </slot>
+
+    <!-- Body -->
+    <section class="modalBody">
+      <slot />
+    </section>
+
+    <slot name="footer">
+      <section class="modalFooter">
+        <!-- Action Buttons -->
+        <div class="text-right">
+          <v-btn
+            v-if="showCancelButton"
+            class="cancelBtn"
+            rounded
+            text
+            @click="$emit('cancel') && $emit('hide')"
+            >{{ buttonCancelText }}</v-btn
+          >
+          <v-btn rounded :disabled="buttonOkDisabled" @click="$emit('ok')">{{
+            buttonOkText
+          }}</v-btn>
+        </div>
+      </section>
+    </slot>
+  </v-dialog>
+</template>
+
+<script>
+export default {
+  props: {
+    header: {
+      type: String,
+      default: ''
+    },
+    imagePath: {
+      type: String,
+      default: ''
+    },
+    show: Boolean,
+    buttonOkText: {
+      type: String,
+      default: 'Ok'
+    },
+    buttonCancelText: {
+      type: String,
+      default: 'Cancel'
+    },
+    buttonOkDisabled: {
+      type: Boolean,
+      default: false
+    },
+    showCancelButton: {
+      type: Boolean,
+      default: true
+    }
+  }
+}
+</script>
+
+<style lang="scss">
```

Comment:
why not scoped

Reply:
Firstly is there any guidlines about scoping styles? Some places are scoped, others a not. Secondly, original component I refactored was not scoped https://github.com/ezl/dailyidea.com/blob/fdeb1b16b0790638ba3213dce76dbcbd8d28fc4c/front/components/dialogs/shareIdeaByEmail.vue#L192

Reply:
There are list of places without scope https://prnt.sc/sh8kf0

Reply:
i know, it should be scoped everywhere its remains of some bad code which we have quite a lot

Reply:
I'll add it here, for other places should we create new issue

Reply:
Ok, added "scoped"

Reply:
I remove scoped cause I don't understand the point of adding shared styles:

```
<style scoped lang="scss">
@import '~assets/style/common';
@import '~assets/style/modals';
```

to every single component that uses dialog. It doesn't work if I only add styles to base dialog component. As I understand it will basically produce lots of duplicates of styles that otherwise are reused. So what's the point of 'scoped' here?